Make unpublishing scary

O-kasso opened the pull request:
Addresses #1718 

- Space unpublish button away from link
- Basic confirm dialog
- my attempt at (English) copy: "Are you sure you want to unpublish this project? Links that have already been shared will no longer work."
- Change icon to trash can
- Change button to red (the color guide doesn't have a dark red, so I just added a 10% darken for hover)
- Opted out of changing "Unpublish" to "Delete" for now, but open to making the switch

Things to address in later PRs:
- Figure out Localization strings
- Replace confirm dialog with prettier

gideonthomas commented on public/editor/scripts/editor/js/fc/publisher.js:
```diff
@@ -173,52 +173,54 @@ define(function(require) {
-      if(request.status !== 200) {
-        console.error("[Thimble] Server was unable to unpublish project, responded with status ", request.status);
-        buttons.unpublish.on("click", handlers.unpublish);
+    if (window.confirm("Are you sure you want to unpublish this project? Links that have already been shared will no longer work.")) {
```
store the result of the confirm dialog in a variable and let's do something like:
```js 
if(!didConfirm) {
  return;
}

...
```

gideonthomas commented on public/editor/scripts/editor/js/fc/publisher.js:
```diff
@@ -173,52 +173,54 @@ define(function(require) {
-        console.error("[Thimble] Server was unable to unpublish project, responded with status ", request.status);
-        buttons.unpublish.on("click", handlers.unpublish);
+    if (window.confirm("Are you sure you want to unpublish this project? Links that have already been shared will no longer work.")) {
+      if (publisher.unpublishing) {
```
This should be outside the confirm `if` check because we do not want to show the confirm dialog if it is already in the process of unpublishing

gideonthomas commented on public/editor/scripts/editor/js/fc/publisher.js:
```diff
@@ -173,52 +173,54 @@ define(function(require) {
+      publisher.unpublishing = true;
 
-      buttons.parent.removeClass("hide");
+      var setState = function(done) {
```
let the name be bound to the function. No need for a variable.

O-kasso replied:
agreed, only added it to avoid a declaration inside an if statement, which I've moved as per your above comments 👍

O-kasso commented:
@gideonthomas I believe I've implemented your feedback :). I've also added the new icon provided [here](https://github.com/mozilla/thimble.mozilla.org/issues/1718#issuecomment-280397148) to `public/resources/img/icon/` -- I noticed a similar folder in `dist/` but I'm not sure what that's for. Should it be copied there as well?

flukeout commented:
Hey @O-kasso checked out the PR, looks good! I've got some suggestions:

**For the garbage can icon, use**
* opacity: 1
* get rid of the ``width`` declaration
* make the ``height`` 16px
* add ``top: 1px`` 

**For the Button**
* Change label to "Delete Published Version"
* User the current, darker, ``:hover`` state as the ``:active`` state.
* For ``:hover`` just set the opacity of the button to .9

**For the dialog**
* Change copy to: "Are you sure? The current link will stop working and if you publish again, the link will be different."

Thanks a lot!

I think down the road, we'll want to move the Confirm copy into the dialog itself. But I'll have to work out the design of that first.

flukeout commented:
Related to my previous comment, here's what I was thinking...

![image](https://cloud.githubusercontent.com/assets/25212/23087417/1b49998c-f529-11e6-8288-9f201c610536.png)

Not needed for this issue. But I think it would be better to put the copy outside of the Confirm, since it's kind of hard to read there. In the confirm we can just do "Are you sure?".

gideonthomas commented on public/editor/scripts/editor/js/fc/publisher.js:
```diff
@@ -173,6 +173,11 @@ define(function(require) {
+    if (!didConfirm) {
+      return;
+    }
     if (publisher.unpublishing) {
```
Can you put this check before the `didConfirm` declaration so that we don't show it if it is already unpublishing.

O-kasso replied:
I believe this has been addressed.

flukeout commented:
Hey @O-kasso - I'm getting a double confirm message when running this patch.

![double-message](https://cloud.githubusercontent.com/assets/25212/23314340/6ac495e6-fa76-11e6-970e-7b786077e594.gif)

Can you update it so that there is just one message (let's use the copy from the second one).

Thanks!

O-kasso commented:
@flukeout I'm unable to repro the double confirm dialogue. I've also tried grepping my branch for the erroneous string with no luck. Do you have the most up-to-date version of the branch?

Pomax commented on public/editor/scripts/editor/js/fc/publisher.js:
```diff
@@ -186,6 +186,13 @@ define(function(require) {
       return;
     }
+
+    var didConfirm = window.confirm("Are you sure? The current link will stop working and if you publish again, the link will be different.");
```
Should this string be localized? It feels like it should be.

gideonthomas replied:
Ah, nice catch! @O-kasso, you can find information about how to localize this here: https://github.com/mozilla/thimble.mozilla.org/wiki/Localization#how-it-works and https://github.com/mozilla/thimble.mozilla.org/wiki/Localization#ui-strings

flukeout commented:
@O-kasso let me know if you have troubles with the localization part. I was working on some stuff that included localization so I'm happy to help.

O-kasso commented:
@flukeout @Pomax I've added localization for en-US and en-GB. What's the process for getting translations in other locales? Is it my responsibility to use Google Translate or something? Right now it'll just print an empty string.

flukeout commented:
Hey awesome @O-kasso it should use the English version as a default if it hasn't yet been localized.

Did you run ``npm run localize`` in your vagrant instance? That's what creates the other locale files. Then you can run ``npm run localize-client`` to uhhh, do something else - at least those are the steps that worked for me.

To run those commands you have to first...

``vagrant ssh``
`` cd /vagrant``

In any case, I'll try this patch out now!

flukeout commented:
Okay - looks like the Reviews have been addressed and this works as expected.

flukeout commented:
Hey @O-kasso I'm collecting information for a new contributors page for the Thimble website and I'd like to include you but need some more details. Can you please email me at ``luke@mozillafoundation.org``? Thanks!
